In [1]:
import pandas as pd

In [2]:
specs = pd.read_csv('data/specs.csv')

## Interesting Strings

* correct
* incorrect
* hint
* did players who received instruction X do better than those who did not
* someplace other than // diagnose player strategies and understanding
* hovers the mouse // determine which objects the player thinks are important
* help button // too difficult
* completion // calculating time spent in a round (for speed and accuracy), attempts at solving a round, and the number of rounds the player has completed
* movie starts to play // used to determine how long players spend watching the movies
* clicks on something that isn't covered elsewhere // attractive distractions, near misses
* exit game event
* delivers instructions to the player // differs from events 3020 and 3021 // determine the effectiveness of the instructions
* beat round event
* skips the tutorial by clicking on the skip button
* done button to submit
* start round event
* indicate a significant change in state during play
* end tutorial event is triggered when the player finishes the tutorial
* finished round event // hit milestone
* clicks on the button to play again
* video starts playing
* start tutorial
* start game event

Example hint events

In [8]:
for st in df[df['info'].str.contains('hint')]['info'].unique():
    print(f'{st}\n')

This event occurs when the player clicks on the character that provides a hint (Sorry, I don't know his name). It contains information about the state of the game when the player chooses to seek help. This event is usually followed by a 3010 (system-initiated instruction) event. It helps identify points that players feel are too difficult.

This event occurs when the player clicks on the character that provides a hint (Sorry I don't know his name). It contains information about the state of the game when the player chooses to seek help. This event is usually followed by a 3010 (system-initiated instruction) event. It helps identify points that players feel are too difficult.

This event occurs when the player clicks on the button that provides a hint. It contains information about the state of the game when the player chooses to seek help. This event is usually followed by a 3010 (system-initiated instruction) event. It helps identify points that players feel are too difficult.



In [9]:
df = df[~df['info'].str.contains('hint')]

## Loop through event code data

In [10]:
train = pd.read_csv('data/train.csv')

In [ ]:
for idx, grp in train.groupby(['event_code']):
    if grp['event_id'].nunique() > 1:
        print(f'Event Code: {idx}')
        for event in grp['event_id'].unique():
            print(f'\tEvent ID: {event}')
            info = specs[specs['event_id'] == event]['info'].iloc[0]
            print(f'\t\t{info}')
        print('\n')

## Event Code

* 2000: start game event at beginning of level, don't wait for intro movie to finish
* 2010: exit game event triggered
* 2020: start round event
* 2025: reset dino, start picking mushroom, start filling tub
* 2030: beat round event, finished round event
* 2035: beat round, finish
* 2060: start tutorial
* 2070: finishes tutorial
* 2075: skip tutorial
* 2080: movie start event, more relevant to first play through when skip is available
* 2081: movie skipped
* 2083: movie ends
* 3010: system-initiated instruction event
* 3020: system-initiated feedback - incorrect events
* 3021: system-initiated feedback - correct events
* 3110: end system-initiated instruction event, useful for amount of time on feedback
* 3120: end system-initiated feedback - incorrect events, useful for how much time on feedback
* 3121: system-initiated feedback - correct events, useful for how much time on feedback
* 4010: player clicks start game from start screen
* 4020: player action in game
* 4021: player action in game
* 4022: player action in game
* 4025: player action in game
* 4030: player action in game
* 4035: player action in game, drag resource to someplace other than acceptable area
* 4040: player action in game
* 4045: player action in game
* 4070: player action in game, clicks something that isn't covered elsewhere, near miss, attractive distraction
* 4080: hovers mouse over interactive object
* 4090: clicks help button (followed by 3010)
* 4095: clicks play again
* 4100: clicks done, submits a solution
* 4110: clicks done, submits a solution, clicks GO during tutorial phase

So there is different feedback and instructions. How does that play into everything? Might check `event_info`.

## Merge train and specs to get event_code

In [ ]:
train = train[train['event_id', 'event_code']]

In [ ]:
enh_specs = specs.merge(train, on='event_id')

In [81]:
enh_specs

event_id                                               info  \
0    2b9272f4  The end of system-initiated feedback (Correct)...   
1    df4fe8b6  The end of system-initiated feedback (Incorrec...   
2    3babcb9b  The end of system-initiated instruction event ...   
3    7f0836bf  The end of system-initiated instruction event ...   
4    ab3136ba  The end of system-initiated instruction event ...   
..        ...                                                ...   
379  29f54413  The start round event is triggered at the star...   
380  06372577  The start tutorial event is triggered at the s...   
381  2a444e03  This event occurs when the player picks up a w...   
382  9e6b7fb5  This event occurs when the player clicks on th...   
383  d3640339  This event occurs when the player clicks on th...   

                                                  args  event_code  
0    [{"name":"game_time","type":"int","info":"mill...        4070  
1    [{"name":"game_time","type":"int","info":"mill...        4030  
2    [{"name":"game_time","type":"int","info":"mill...        4030  
3    [{"name":"game_time","type":"int","info":"mill...        4070  
4    [{"name":"game_time","type":"int","info":"mill...        4030  
..                                                 ...         ...  
379  [{"name":"game_time","type":"int","info":"mill...        4035  
380  [{"name":"game_time","type":"int","info":"mill...        4030  
381  [{"name":"game_time","type":"int","info":"mill...        4030  
382  [{"name":"game_time","type":"int","info":"mill...        4030  
383  [{"name":"game_time","type":"int","info":"mill...        3010  

[384 rows x 4 columns]